src/Datadog.Trace/Util/DomainMetadata.cs

```diff
@@ -9,7 +9,10 @@ namespace Datadog.Trace.Util
     internal static class DomainMetadata
     {
         private const string UnknownName = "unknown";
-        private static Process _currentProcess;
+        private static bool _initialized;
+        private static string _currentProcessName;
+        private static string _currentProcessMachineName;
+        private static int _currentProcessId;
         private static bool _processDataPoisoned;
```

Comment:
Do we need `_processDataPoisoned`? Can the name or the id of the current process ever change? Or the machine name?

Reply:
@colin-higgins what do you think?

Reply:
Looking at the code again after a good night's sleep, it looks like this flag just means "we already tried to get the process name, id, etc once, but that failed, so don't try again." From the field name I thought it was used to indicate that the data had changed and we needed to retrieve it again (aka "dirty").

Maybe we should rename it to `_processDataNotAvailable`?

Reply:
Yeah that makes sense to me. Implemented in 46dac65e6b5e57045338b9adf09943df8327e112